src/ShallowWrapper.js

```diff
@@ -346,16 +347,22 @@ export default class ShallowWrapper {
    * @param {Array} args
    * @returns {ShallowWrapper}
    */
-  simulate(event, ...args) {
-    const handler = this.prop(propFromEvent(event));
-    if (handler) {
-      withSetStateAllowed(() => {
-        // TODO(lmr): create/use synthetic events
-        // TODO(lmr): emulate React's event propagation
-        handler(...args);
-        this.root.update();
-      });
-    }
+  simulate(event, mock, ...args) {
+    const eventProp = propFromEvent(event);
+    const e = Object.assign(new SyntheticEvent(undefined, undefined, { type: event }), mock);
+    withSetStateAllowed(() => {
+      let current = this;
```

Comment:
thats the variable that keeps getting updated as i traverse the tree from the event's target up until the root, i should probbaly make this a for loop to make it clearer